sympy/concrete/tests/test_sums_products.py

```diff
@@ -1216,7 +1216,7 @@ def test_Sum_dummy_eq():
 
 
 def test_issue_15852():
-    assert summation(x**y*y, (y, -oo, oo)).doit() == Sum(x**y*y, (y, -oo, oo))
+    assert summation(x**y*y, (y, -oo, oo)).doit() == Sum(x**y*y, (y, -oo, oo)), summation(x**y*y, (y, -oo, oo)).doit()
```

Comment:
```suggestion
    # unevaluated or Piecewise in terms of expression that should evaluate
    # to unevaluated summation but doesn't because of issue 23914
    assert summation(x**y*y, (y, -oo, oo)).doit() != Dummy('hang')
```